CauseContract.sol

```diff
@@ -52,20 +53,25 @@ contract CauseContract {
         return ContractInfo(id, admin, incoming, outgoing, contractAddress);
     }
 
-    function donate() public payable {
+    uint256 public transactionFee;
+    
+    function donate() public payable returns (uint256) {
         require(msg.value > 0, "You must send some Ether");
 
-        incoming.push(Transaction(msg.sender, msg.value * (100 - feePercent) / 100, block.timestamp, block.number, tx.gasprice, 2));
+        transactionFee = (msg.value * tx.gasprice * BASIS_POINTS) / 10000; // Transaction fee of 50bps (by default)
+        incoming.push(Transaction(msg.sender, msg.value - transactionFee, block.timestamp, block.number, tx.gasprice, transactionFee));
 
-        blockchange.transfer(msg.value * feePercent / 100);
+        blockchange.transfer(transactionFee);
+        return transactionFee; // Fee in Wei
```

Comment:
Since Solidity doesn't deal with decimals, I find it hard to convert the unit from Wei to ETH here. 
I believe the unit conversion may be handled quickly at the front end.

Reply:
Yup I agree that will make it easier, I'll merge this PR for our new sprint tomorrow